assets/js/components/legacy-notifications/notification.js

```diff
@@ -22,68 +22,87 @@
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import { map } from 'lodash';
-/**
+
+/*
  * WordPress dependencies
  */
-import { Component, Fragment, createRef, isValidElement } from '@wordpress/element';
+import { useState, useRef, useEffect, Fragment, isValidElement } from '@wordpress/element';
 
-/**
+/*
  * Internal dependencies
  */
 import GoogleLogoIcon from '../../../svg/logo-g.svg';
 import { sanitizeHTML } from '../../util/sanitize';
-import { setCache, getCache, deleteCache } from '../data/cache';
 import DataBlock from '../DataBlock';
 import Button from '../Button';
 import Warning from './warning';
 import Error from './error';
 import Link from '../Link';
 import ModuleIcon from '../ModuleIcon';
-
-class Notification extends Component {
-	constructor( props ) {
-		super( props );
-
-		this.state = {
-			isClosed: false,
-		};
-
-		this.cardRef = createRef();
-
-		this.handleDismiss = this.handleDismiss.bind( this );
-		this.handleCTAClick = this.handleCTAClick.bind( this );
-
-		if ( 0 < this.props.dismissExpires ) {
-			this.expireDismiss();
+import { getItem, setItem, deleteItem } from '../../googlesitekit/api/cache';
+
+function Notification( {
```

Comment:
The previous class component was calling `getCache` in its constructor. Since the cache methods are now async, it needed a workaround.
I discussed with @tofumatt and we decided to just move it over to a functional component and use hooks.